### PR TITLE
Added additional S3 Discovery instructions

### LINKED ISSUE #8351: AWS S3 Discovery processor not added
## Expected Behavior
   
Going through the documents here : https://backstage.io/docs/integrations/aws-s3/discovery it is assumed that there are no additional config steps required in order to enable S3 discovery. I have an S3 integration added, and can add items from my bucket into the catalog using the importer, but discovery would throw a warning message: `warn No processor was able to handle reading of s3-discovery:https://my-bucket.s3.ap-southeast-2.amazonaws.com/` 

## Current Behavior

s3-discovery processor not added and a  `warn No processor was able to handle reading of s3-discovery:https://my-bucket.s3.ap-southeast-2.amazonaws.com/` message is raised.

## Possible Solution

If I add a new processor in plugins/catalog.ts then the s3 discovery works fine :
`builder.addProcessor(new AwsS3DiscoveryProcessor(reader))`
 
I'm not sure if this is expected, or if there is just a gap in the documentation right now .

## Steps to Reproduce

Create S3 bucket and provide access
 
Add AWS S3 Integration:
https://backstage.io/docs/integrations/aws-s3/locations#configuration
 
Add s3 discovery location:
https://backstage.io/docs/integrations/aws-s3/discovery
 
No processor warning for s3-discovery : ` warn No processor was able to handle reading of s3-discovery:https://my-bucket.s3.ap-southeast-2.amazonaws.com/`

## Context

Trying to utilise the s3 discovery to import all my entities

## Your Environment

OS:   Darwin 20.6.0 - darwin/arm64
node: v14.18.2
yarn: 1.22.10
cli:  0.9.1 (installed)

### PR DESCRIPTION
**Pull request by nigelwright-roam**
Signed-off-by: Nigel Wright <nigel.wright@roamdigital.com>

## Hey, I just made a Pull Request!

Updated the s3 discovery document to include information on how to add the processor. Fixes #8351 
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/integrations/aws-s3/discovery.md [hunk 1]
```diff
@@ -26,3 +26,16 @@ catalog:
 ```
 
 Note the `s3-discovery` type, as this is not a regular `url` processor.
+
+As this processor is not one of the default providers, you will also need to add the below:
```
Comment (Rugvip) [inline docs/integrations/aws-s3/discovery.md:30]: Perhaps "... will also need to add it to `packages/backend/src/plugins/catalog.ts`:"?

File: docs/integrations/aws-s3/discovery.md [hunk 2]
```diff
@@ -26,3 +26,16 @@ catalog:
...
+
+```javascript
```
Comment (Rugvip) [inline docs/integrations/aws-s3/discovery.md:32]: no big deal, but `ts` makes more sense here :grin:

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍

File: docs/integrations/aws-s3/discovery.md (hunk 2 above)
Comment (nigelwright-roam) [reply docs/integrations/aws-s3/discovery.md:32]: true - updated now!

File: docs/integrations/aws-s3/discovery.md (hunk 1 above)
Comment (nigelwright-roam) [reply docs/integrations/aws-s3/discovery.md]: makes sense to clarify location earlier than the code snippet :)

**Comment by benjdlambert** [comment]
@nigelwright-roam think there's some failures with prettier in CI. You might want to run prettier on the changes files to get them to format the `.md` correctly 👍

**Comment by nigelwright-roam** [comment]
@benjdlambert - sorry about that! Ran prettier on the file, hopefully is all ok now.

**Review by freben — APPROVED** [review]

**Comment by suuus** [comment]
Thanks for your contribution @nigelwright-roam 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**